partiql-eval/src/main/kotlin/org/partiql/eval/internal/helpers/ToBag.kt

```diff
@@ -12,7 +11,14 @@ import org.partiql.value.PartiQLValueExperimental
  *  - F(array_value)  -> bag_value          # discard ordering
  *  - F(bag_value)    -> bag_value          # identity
  */
-@OptIn(PartiQLValueExperimental::class)
-internal fun PartiQLValue.toBag(): BagValue<*> {
-    TODO("For OUTER set operators")
+internal fun Datum.asIterator(): Iterator<Datum> {
```

Comment:
Based on an offline discussion w/ @johnedquinn, mistyped cases in strict mode should result in an error.

For reference:
- partiql-lang issue https://github.com/partiql/partiql-lang/issues/76
- partiql-tests fix https://github.com/partiql/partiql-tests/pull/120/files

I will correct this in a new commit